third_party/nvidia/backend/compiler.py

```diff
@@ -106,6 +106,13 @@ def parse_options(self, opts) -> Any:
         args["max_num_imprecise_acc_default"] = 2**30 if self.capability == 90 else 0
         return CUDAOptions(**args)
 
+    def get_codegen_implementation(self):
+        import triton.language.extra.cuda as cuda
+        codegen_fn = dict()
+        codegen_fn[
+            "convert_custom_types"] = cuda.convert_custom_float8_sm90 if self.capability >= 90 else cuda.convert_custom_float8_sm80
```

Comment:
I'm not sure about the name "convert_custom_types", because actually this function's signature is

```
convert_custom_float8(arg, dst_ty, fp_downcast_rounding, has_minx2, _builder=None)
```

If it was just (arg, dst_ty) then I could see how it's generic, but with the other two arguments it doesn't feel generic to me.

Reply:
the signature is actually (there are no has_minx2):
```
convert_custom_float8(arg, dst_ty, fp_downcast_rounding, _builder=None)
```

so should be generic for any type conversion right?

Reply:
Any generic **floating-point** type conversion, yeah.